core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2727,31 +2791,37 @@ public void unmount(AlluxioURI alluxioPath) throws FileDoesNotExistException,
    *
    * @param rpcContext the rpc context
    * @param inodePath the Alluxio path to unmount, must be a mount point
+   * @param removeInodes whether to remove inodes after mount point is deleted
    */
-  private void unmountInternal(RpcContext rpcContext, LockedInodePath inodePath)
-      throws InvalidPathException, FileDoesNotExistException, IOException {
-    MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
-    mSyncManager.stopSyncForMount(resolution.getMountId());
-
+  private void unmountInternal(RpcContext rpcContext, LockedInodePath inodePath,
+      boolean removeInodes) throws InvalidPathException, FileDoesNotExistException, IOException {
     if (!inodePath.fullPathExists()) {
       throw new FileDoesNotExistException(
           "Failed to unmount: Path " + inodePath.getUri() + " does not exist");
     }
+    MountInfo mountInfo = mMountTable.getMountTable().get(inodePath.getUri().getPath());
+    if (mountInfo == null) {
+      throw new InvalidPathException("Failed to unmount " + inodePath.getUri() + ". Please ensure"
+          + " the path is an existing mount point.");
+    }
+    mSyncManager.stopSyncForMount(mountInfo.getMountId());
```

Comment:
Does this mean if the user updates mount for a previously active synced mount point, they will need to set up the active sync again?

Reply:
yeah looks like this command will remove all the sync points under the mount point. I can try adding them back after the remount is completed.